lib/matplotlib/axes/_axes.py

```diff
@@ -1544,39 +1544,43 @@ def loglog(self, *args, **kwargs):
     def semilogx(self, *args, **kwargs):
         """
         Make a plot with log scaling on the *x* axis.
+        
+        Parameters
+        ----------
+        basex : float, optional
+            Base of the *x* logarithm.
 
-        Call signature::
-
-          semilogx(*args, **kwargs)
-
-        :func:`semilogx` supports all the keyword arguments of
-        :func:`~matplotlib.pyplot.plot` and
-        :meth:`matplotlib.axes.Axes.set_xscale`.
-
-        Notable keyword arguments:
-
-          *basex*: scalar > 1
-            Base of the *x* logarithm
-
-          *subsx*: [ *None* | sequence ]
+        subsx : array_like, optional
             The location of the minor xticks; *None* defaults to
             autosubs, which depend on the number of decades in the
             plot; see :meth:`~matplotlib.axes.Axes.set_xscale` for
             details.
 
-          *nonposx*: [ 'mask' | 'clip' ]
+        nonposx : string, optional, {'mask', 'clip'}
             Non-positive values in *x* can be masked as
-            invalid, or clipped to a very small positive number
+            invalid, or clipped to a very small positive number.
 
-        The remaining valid kwargs are
+        Returns
+        -------
+        `~matplotlib.pyplot.plot`
+            Log-scaled plot on the *x* axis.
+
+        Other Parameters
+        ----------------
         :class:`~matplotlib.lines.Line2D` properties:
 
         %(Line2D)s
```

Comment:
`Line2D` will need to be updated in order for this interpolation to work correctly.